### DISCUSSION TITLE
Supporting a new major version

### DISCUSSION THREAD

**Discussion by eyalbe4** [Homebrew/discussions#1688] [Tap maintenance and Homebrew development]
We're planning to release a new major version (v2) for jfrog-cli (https://github.com/Homebrew/homebrew-core/blob/master/Formula/jfrog-cli.rb).
Since this version will include breaking changes, we thought it would be good to create a new formula for it, once the version is available. This way, users who install jfrog-cli with brew will not receieve the new version automatically.
The v2 code will be maintained in a new branch under https://github.com/jfrog/jfrog-cli
Are there any recommendations for us to follow to implement this?
Appreciate your help with this.

**Comment by SMillerDev** [comment]
> Since this version will include breaking changes, we thought it would be good to create a new formula for it, once the version is available. This way, users who install jfrog-cli with brew will not receieve the new version automatically.

This is against the homebrew versioning policy. Unversioned formulae should always be the latest upstream version.

> The v2 code will be maintained in a new branch under https://github.com/jfrog/jfrog-cli
Are there any recommendations for us to follow to implement this?

If there is a plan in place to keep supporting the old version you could make a jfrog-cli@1 formula. See https://docs.brew.sh/Versions for all requirements there.

**Reply by eyalbe4** [reply]
Thabks for your response @SMillerDev!
We'll go with the jfrog-cli@2 formula for v2 of jfrog-cli.

And a follow up question - 
We'd like make sure no one updates the v1 formula, to install a new v2 version. Is the recommended way to protect against this, is by adding (or modifying) the test inside the formula?

**Reply by SMillerDev** [reply]
>  We'll go with the jfrog-cli@2 formula for v2 of jfrog-cli.

That's not what I said. `jfrog-cli` (the formula without a version in the name) MUST be the latest upstream version according to Homebrew's versioning policy. You can make a `jfrog-cli@1` formula if it meets all requirements.

> We'd like make sure no one updates the v1 formula, to install a new v2 version. Is the recommended way to protect against this, is by adding (or modifying) the test inside the formula?

I think maintainers will catch it if someone updates `jfrog-cli@1` to a version that doesn't start with `1.x`

**Reply by eyalbe4** [reply]
Got it @SMillerDev.
Is there a way to protect against the following situation?
People who have automation scripts that include ```brew install jfrog-cli``` automatically get upgraded to v2, which includes the breaking changes?

**Reply by SMillerDev** [reply]
Write a blogpost announcing that V2 is out. If these people know anything about the system they automated they'll know that Homebrew will be shipping v2 as soon as possible after the official release.
